test/index.js

```diff
@@ -1117,6 +1118,60 @@ describe('list', function() {
 
   });
 
+  describe('slice', function() {
+
+    it('returns a Nothing if applied to an empty list', function() {
+      assert(S.slice(0, 3, []).equals(S.Nothing()));
+    });
+
+    it('returns a Nothing with a positive end index greater than start index', function() {
+        assert(S.slice(6, 1, [1, 2, 3, 4, 5]).equals(S.Nothing()));
+      });
+
+    it('returns a Nothing with a positive end index greater than list length', function() {
+        assert(S.slice(1, 6, [1, 2, 3, 4, 5]).equals(S.Nothing()));
+      });
+
+    it('returns a Nothing with a negative end index greater than list length', function() {
+        assert(S.slice(1, -6, [1, 2, 3, 4, 5]).equals(S.Nothing()));
+      });
+
+    it('returns a Nothing with a negative start index greater than list length', function() {
+      assert(S.slice(-6, 1, [1, 2, 3, 4, 5]).equals(S.Nothing()));
+    });
+
+    it('returns a Just with an empty array when start index equals end index', function() {
+        eq(S.slice(1, 1, [1, 2, 3, 4, 5]).toString(), 'Just([])');
```

Comment:
This could be expanded to eight assertions:

```javascript
eq(S.slice(1, 1, [1, 2, 3, 4, 5]).toString(), 'Just([])');
eq(S.slice(1, -4, [1, 2, 3, 4, 5]).toString(), 'Just([])');
eq(S.slice(-4, 1, [1, 2, 3, 4, 5]).toString(), 'Just([])');
eq(S.slice(-4, -4, [1, 2, 3, 4, 5]).toString(), 'Just([])');
eq(S.slice(0, 0, []).toString(), 'Just([])');
eq(S.slice(0, -0, []).toString(), 'Just([])');
eq(S.slice(-0, 0, []).toString(), 'Just([])');
eq(S.slice(-0, -0, []).toString(), 'Just([])');
```